decode.go

```diff
@@ -68,13 +68,19 @@ func (dec *Decoder) ScanKeyval() bool {
 	return false
 
 key:
+	const invalidKeyError = "invalid key"
+
 	start := dec.pos
 	for p, c := range line[dec.pos:] {
 		switch {
 		case c == '=':
 			dec.pos += p
 			if dec.pos > start {
 				dec.key = line[start:dec.pos]
+				if invalidKey(dec.key) {
+					dec.syntaxError(invalidKeyError)
+					return false
+				}
```

Comment:
We can avoid doing extra work for ASCII only keys.

1. Initialize `multibyte := false` outside the loop.
2. Add `case c >= utf8.RuneSelf:	multibyte = true` at the bottom of the switch.
3. Change each `if invalidKey(dec.key)` to `if multibyte && bytes.IndexRune(dec.key, utf8.RuneError) != -1`